### PR TITLE
cleanup: remove unused buildpack

### PR DESCRIPTION
**Pull request by coryan**
We used to create a buildpack for the CI builds, and a separate
buildpack for the user-facing READMEs. A while back we changed the
user-facing READMEs to use the official GCP buildpacks, this cleans up
the now (largely) unused code.

I also updated the documents that referenced these buildpacks when they
shouldn't, and I took the opportunity to (a) clarify what is the purpose
and intended audience of these docs, (b) cross-reference the docs in
case folks land in the wrong document, and (c) create an index of the
key how-to guides to have a single thing to cross-reference.

### CODE REVIEW

File: ci/README.md [hunk 1]
```diff
@@ -1,14 +1,19 @@
 # Configuring the CI Environment
 
-We use several Google Cloud Platform services to perform integration tests.
-These notes describe how to setup a project to run these integration tests.
-The main audience for these notes are developers working to improve the
-Functions Framework for C++. They assume the reader is familiar with GCP and
-with the Google Cloud SDK command-line tool.
+These notes describe how to set up a GCP project to run these Functions Framework for C++ integration tests. The main
```
Comment (dbolduc) [inline ci/README.md:3]: nit: 80 character limit?

File: ci/README.md [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+audience for these notes are developers of the Functions Framework.
+
+- If you are looking for examples on how to **use** the Functions Framework for C++, the
+  [How-to Guides](/examples/howto-guides.md) should help you get started.
+- If you want to compile and test the examples after making changest to the framework, the read
```
Comment (dbolduc) [inline ci/README.md:8]: s/changest/changes/
s/the read/then read/

File: ci/README.md [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+  [Testing the Functions Framework Examples](/examples/README.md)
+- If you are looking for more detailed instructions to configure the CI environment, this is the document you should
```
Comment (dbolduc) [inline ci/README.md:10]: s/more detailed/detailed/

File: examples/README.md [hunk 4]
```diff
@@ -1,31 +1,39 @@
-# Functions Framework Examples
+# Testing the Functions Framework Examples
 
-This directory contains a series of examples showing how to use the Functions Framework.
+These notes describe how build and tests these examples when developing the Functions Framework for C++. The main
```
Comment (dbolduc) [inline examples/README.md:3]: s/how build and tests/how to build and test/

File: examples/howto-guides.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# How-to Guides: Index
+
+* [Running your function as Docker container](/examples/site/howto_create_container/README.md) building and running your
```
Comment (dbolduc) [inline examples/howto-guides.md:3]: For all of these bullets, consider some sort of punctuation after the hyperlink. Maybe: `[Title](link) - building and running...`

**Review by dbolduc — APPROVED** [review]

File: examples/README.md (hunk 4 above)
Comment (coryan) [reply examples/README.md:3]: Done.

File: ci/README.md (hunk 3 above)
Comment (coryan) [reply ci/README.md:10]: Done.

File: ci/README.md (hunk 2 above)
Comment (coryan) [reply ci/README.md:8]: Done.

File: ci/README.md (hunk 1 above)
Comment (coryan) [reply ci/README.md:3]: Done.

**Pull request merged**